backend/src/main/scala/bloop/BloopClassFileManager.scala

```diff
@@ -180,14 +180,21 @@ final class BloopClassFileManager(
       // invalidations can happen in both paths, no-op if missing
       allInvalidatedClassFilesForProject.-=(generatedClassFile)
       allInvalidatedClassFilesForProject.-=(rebasedClassFile)
-      supportedCompileProducts.foreach { supportedProductSuffix =>
-        val productName = rebasedClassFile
+
+      def productFile(classFile: File, supportedProductSuffix: String) = {
+        val productName = classFile
           .getName()
           .stripSuffix(".class") + supportedProductSuffix
-        val productAssociatedToClassFile =
-          new File(rebasedClassFile.getParentFile, productName)
-        if (productAssociatedToClassFile.exists())
-          allInvalidatedExtraCompileProducts.-=(productAssociatedToClassFile)
+        new File(classFile.getParentFile, productName)
+      }
+      supportedCompileProducts.foreach { supportedProductSuffix =>
+        val generatedProductName = productFile(generatedClassFile, supportedProductSuffix)
+        val rebasedProductName = productFile(rebasedClassFile, supportedProductSuffix)
+
+        if (generatedProductName.exists() || rebasedProductName.exists()) {
```

Comment:
I don't think it is useful to check if they exists. It is okay to remove a element from a set even if it is not there.

Reply:
I agree, but it breaks another test :sweat_smile: 

There is a test that creates a fake nir files and expects them to be removed. So if you change the target from scalaJs to jvm it should remove the old files.